rllib/algorithms/impala/impala.py

```diff
@@ -938,35 +938,37 @@ def learn_on_processed_samples(self) -> ResultDict:
             Aggregated results from the learner group after an update is completed.
 
         """
-        result = {}
-        # There are batches on the queue -> Send them to the learner group.
+        # There are batches on the queue -> Send them all to the learner group.
         if self.batches_to_place_on_learner:
-            batch = self.batches_to_place_on_learner.pop(0)
+            batches = self.batches_to_place_on_learner[:]
```

Comment:
Per iteration, we now purge all accumulated training batches and send them - at once - to the LearnerGroup.